### PR TITLE
cmake: Update Emscripten.cmake to be aware of new sysroot location.

### LINKED ISSUE #6595: emcmake find_package not working for own libraries
I hope this is the correct place to ask this question.

I have a library Foo which I install and a program Bar which requires this library.

In my library foo I create a FooConfig:
`
include("${CMAKE_CURRENT_LIST_DIR}/FooTargets.cmake")
`

Which imports the targets I generated in my CMakeLists.txt.

```
install(TARGETS Foo EXPORT FooTargets
	ARCHIVE DESTINATION lib
	LIBRARY DESTINATION lib
	RUNTIME DESTINATION bin
	INCLUDES DESTINATION include
)
install(DIRECTORY include/foo DESTINATION include)

install(EXPORT FooTargets
	FILE FooTargets.cmake
	NAMESPACE Foo::
	DESTINATION lib/cmake/Foo
)

include(CMakePackageConfigHelpers)
write_basic_package_version_file("FooConfigVersion.cmake"
	VERSION ${Foo_VERSION}
	COMPATIBILITY SameMajorVersion
)

install(FILES
	"FooConfig.cmake"
	"${CMAKE_CURRENT_BINARY_DIR}/FooConfigVersion.cmake"
	DESTINATION lib/cmake/Foo
)
```

When I use simply cmake, I can run `find_package(Foo REQUIRED)` in the CMakeLists.txt of Bar,
and the library will be imported.

This however does not work if I run `emcmake cmake ..`. I get the following error message:

```
CMake Error at CMakeLists.txt:8 (find_package):
  By not providing "FindFoo.cmake" in CMAKE_MODULE_PATH this project has
  asked CMake to find a package configuration file provided by "Foo", but
  CMake did not find one.

  Could not find a package configuration file provided by "Foo" with any of
  the following names:

    FooConfig.cmake
    foo-config.cmake

  Add the installation prefix of "Foo" to CMAKE_PREFIX_PATH or set
  "Foo_DIR" to a directory containing one of the above files.  If "Foo"
  provides a separate development package or SDK, be sure it has been
  installed.
```

### PR DESCRIPTION
**Pull request by sbc100**
Also, add more testing in order to verify that finding resources
inside the sysroot works as expected.

See: https://github.com/emscripten-core/emsdk/issues/824#
See: #6595

I'm not sure if this is also fixes #6595 and will likely followup with a
test for find_package.

Fixes: #10078

### CODE REVIEW

File: cmake/Modules/Platform/Emscripten.cmake [hunk 1]
```diff
@@ -211,6 +209,20 @@ if (EMSCRIPTEN_FORCE_COMPILERS)
   endif()
 endif()
 
+execute_process(COMMAND "${EMSCRIPTEN_ROOT_PATH}/em-config${EMCC_SUFFIX}" "CACHE"
+  RESULT_VARIABLE _emcache_result
+  OUTPUT_VARIABLE _emcache_output
+  OUTPUT_STRIP_TRAILING_WHITESPACE)
+if (NOT _emcache_result EQUAL 0)
+  message(FATAL_ERROR "Failed to find emscripten cache directory command \"'${EMSCRIPTEN_ROOT_PATH}/em-config${EMCC_SUFFIX}' CACHE\"! Process returned with error code ${_emcache_result}.")
```
Comment (kripken) [inline cmake/Modules/Platform/Emscripten.cmake:217]:
```suggestion
  message(FATAL_ERROR "Failed to find emscripten cache directory using command: \"'${EMSCRIPTEN_ROOT_PATH}/em-config${EMCC_SUFFIX}' CACHE\"! Process returned with error code ${_emcache_result}.")
```

File: tests/test_core.py [hunk 2]
```diff
@@ -6222,6 +6222,10 @@ def line_splitter(data):
         };
         """ % line_splitter(str(image_bytes)))
 
+      # ensure libpng is built so that openjpeg's configure step can detect it.
+      # If we don't do this then this test will produce different results depending on the current
+      # state of the sysroot.
```
Comment (kripken) [inline tests/test_core.py:6227]:
```suggestion
      # state of the sysroot (as determined by other tests).
```

File: tests/test_other.py [hunk 3]
```diff
@@ -660,6 +660,11 @@ def test_cmake_bitcode_static_libraries(self):
   def test_cmake_emscripten_version(self):
     self.run_process([EMCMAKE, 'cmake', test_file('cmake/emscripten_version')])
 
+  def test_cmake_find_stuff(self):
+    # Ensure that zlib exists in the sysroot
+    self.run_process([EMCC, test_file('hello_world.c'), '-sUSE_ZLIB'])
+    self.run_process([EMCMAKE, 'cmake', test_file('cmake/find_stuff')])
```
Comment (kripken) [inline tests/test_other.py:666]: How about adding this new test to run on windows too?

File: tests/test_core.py (hunk 2 above)
Comment (sbc100) [reply tests/test_core.py:6227]: Reworded.

File: tests/test_other.py (hunk 3 above)
Comment (sbc100) [reply tests/test_other.py:666]: Oh we already run `other.test_cmake*` on windows!

**Review by kripken — APPROVED** [review]

**Pull request merged**